## Getting started

Start by loading the appropriate modules:
```
module use /cvmfs/soft.mugqic/CentOS6/modulefiles
module load mugqic/java mugqic/fastqc
```

Copy the data that you need:
```
rm -r test_data
mkdir test_data
cd test_data
cp /home/etu40/test_data/* .
```

Run the tool fastqc:
```
fastqc ENCFF361CSC_2M_ctl.fastq.gz
fastqc ENCFF837BCE_2M_ctcf.fastq.gz
```

If you do `ls` you should see the resulting .html and .zip files.

After that in a different terminal window on your local machine:
```
scp etuXX@ip31.ccs.usherbrooke.ca:/home/etuXX/test_data/*.html .
```

Replacing the 'etuXX' by your actual account number. 

After that you should be able to open the .html files locally on your browser to look at the result of fastqc.

## Alignment

We will be using the mapping tool bwa.
```
module load mugqic/bwa
```

Create shortcut for human reference sequence:
```
export hg19_ref=/cvmfs/soft.mugqic/CentOS6/genomes/species/Homo_sapiens.hg19/genome/bwa_index/Homo_sapiens.hg19.fa
```

The actual mapping command:
```
bwa aln -t 4 $hg19_ref ENCFF837BCE_2M_ctcf.fastq.gz > ENCFF837BCE_2M_ctcf.sai
```

Can you guess what `-t 4` means?

Next step is to create the actual alignement file:
```
bwa samse -f ENCFF837BCE_2M_ctcf.sam $hg19_ref ENCFF837BCE_2M_ctcf.sai ENCFF837BCE_2M_ctcf.fastq.gz 
```

At this point, doing `ls -ltr` should produce something like:
```
-rwxr-xr-x 1 etu21 etu  94546752 Nov  1 21:53 ENCFF361CSC_2M_ctl.fastq.gz
-rwxr-xr-x 1 etu21 etu  94485769 Nov  1 21:53 ENCFF837BCE_2M_ctcf.fastq.gz
-rw-r--r-- 1 etu21 etu    365149 Nov  1 21:53 ENCFF361CSC_2M_ctl_fastqc.zip
-rw-r--r-- 1 etu21 etu    326714 Nov  1 21:53 ENCFF361CSC_2M_ctl_fastqc.html
-rw-r--r-- 1 etu21 etu    349567 Nov  1 21:54 ENCFF837BCE_2M_ctcf_fastqc.zip
-rw-r--r-- 1 etu21 etu    317492 Nov  1 21:54 ENCFF837BCE_2M_ctcf_fastqc.html
-rw-r--r-- 1 etu21 etu  66628924 Nov  1 22:15 ENCFF837BCE_2M_ctcf.sai
-rw-r--r-- 1 etu21 etu 568344066 Nov  1 22:17 ENCFF837BCE_2M_ctcf.sam
```

Let's compress the .sam file using samtools:
```
module load mugqic/samtools
samtools view -Sb ENCFF837BCE_2M_ctcf.sam > ENCFF837BCE_2M_ctcf.bam 
rm ENCFF837BCE_2M_ctcf.sam
```
Let's look at the file:
```
samtools view ENCFF837BCE_2M_ctcf.bam | less
```

Now let's sort the bam file and look at the results:
```
samtools sort ENCFF837BCE_2M_ctcf.bam -o ENCFF837BCE_2M_ctcf_sorted.bam
samtools view ENCFF837BCE_2M_ctcf_sorted.bam | less
```

Do you understand this file? Where can you find more information about the file format?

Try:
```
samtools flagstat ENCFF837BCE_2M_ctcf_sorted.bam
```

Now, can you actually look the resulting file?

You'll probably need to create an index file...
```
samtools index ENCFF837BCE_2M_ctcf_sorted.bam
```

And a coverage file for IGV:
```
module load mugqic/igvtools
igvtools count ENCFF837BCE_2M_ctcf_sorted.bam ENCFF837BCE_2M_ctcf_sorted.bam.tdf hg19
```

## Redo the analysis using all the data on chromosome 22

Copy the data that you need:
```
cd ..
rm -r test_data2
mkdir test_data2
cd test_data2
cp /home/etu40/test_data2/* .
```

*Complete all other steps...*

Finally, you should get something like this if you do `ls -ltr`:
```
-rwxr-xr-x 1 etu22 etu  8219393 Nov  2 16:41 ENCFF361CSC.chr22_ctl.fastq.gz
-rwxr-xr-x 1 etu22 etu 11297870 Nov  2 16:41 ENCFF837BCE.chr22_ctcf.fastq.gz
-rw-r--r-- 1 etu22 etu   288627 Nov  2 16:45 ENCFF361CSC.chr22_ctl_fastqc.zip
-rw-r--r-- 1 etu22 etu   258845 Nov  2 16:45 ENCFF361CSC.chr22_ctl_fastqc.html
-rw-r--r-- 1 etu22 etu   387398 Nov  2 16:45 ENCFF837BCE.chr22_ctcf_fastqc.zip
-rw-r--r-- 1 etu22 etu   323809 Nov  2 16:45 ENCFF837BCE.chr22_ctcf_fastqc.html
-rw-r--r-- 1 etu22 etu  8124788 Nov  2 16:46 ENCFF837BCE.chr22_ctcf.sai
-rw-r--r-- 1 etu22 etu  5750580 Nov  2 16:47 ENCFF361CSC.chr22_ctl.sai
-rw-r--r-- 1 etu22 etu 14127117 Nov  2 16:49 ENCFF837BCE.chr22_ctcf.bam
-rw-r--r-- 1 etu22 etu 10201844 Nov  2 16:49 ENCFF361CSC.chr22_ctl.bam
-rw-r--r-- 1 etu22 etu 14047177 Nov  2 16:50 ENCFF837BCE.chr22_ctcf_sorted.bam
-rw-r--r-- 1 etu22 etu 10151948 Nov  2 16:50 ENCFF361CSC.chr22_ctl_sorted.bam
-rw-r--r-- 1 etu22 etu  1591112 Nov  2 16:51 ENCFF837BCE.chr22_ctcf_sorted.bam.bai
-rw-r--r-- 1 etu22 etu  1558112 Nov  2 16:51 ENCFF361CSC.chr22_ctl_sorted.bam.bai
-rw-r--r-- 1 etu22 etu  3942604 Nov  2 16:52 ENCFF837BCE.chr22_ctcf_sorted.bam.tdf
-rw-r--r-- 1 etu22 etu  3470484 Nov  2 16:52 ENCFF361CSC.chr22_ctl_sorted.bam.tdf
```

## Calling peaks

At some point you should be able to call peaks, using something like:
```
export MUGQIC_INSTALL_HOME=/cvmfs/soft.mugqic/CentOS6
module use $MUGQIC_INSTALL_HOME/modulefiles
module load mugqic/python/2.7.13

module load mugqic/MACS2
macs2 callpeak --format BAM --nomodel --treatment ENCFF837BCE.chr22_ctcf_sorted.bam --control ENCFF361CSC.chr22_ctl_sorted.bam --name CTCF_input
```

Now if you do `ls -ltr CTCF_input*` you should get:
```
-rw-r--r-- 1 etu22 etu 36956 Nov  2 16:57 CTCF_input_peaks.xls
-rw-r--r-- 1 etu22 etu 32096 Nov  2 16:57 CTCF_input_peaks.narrowPeak
-rw-r--r-- 1 etu22 etu 21129 Nov  2 16:57 CTCF_input_summits.bed
```

## Looking for motifs

We will use a script from a tool call Homer:
```
export MUGQIC_INSTALL_HOME=/cvmfs/soft.mugqic/CentOS6
module load mugqic/perl/5.22.1 mugqic/weblogo/2.8.2 mugqic/homer/4.7
findMotifsGenome.pl CTCF_input_peaks.narrowPeak hg19 peakAnalysis -preparsedDir preparsed -p 8
```

## Looking at overlap with repeats

Retrieve repeatmasker file from the UCSC Table Browser in bed format and save file as `hg19_repeats.bed.gz`

![Table Browser](/table_browser.jpeg)

Copy file to your student account on the cluster:
```
scp hg19_repeats.bed.gz etuXX@ip31.ccs.usherbrooke.ca:/home/etuXX/test_data2/.
```

Where you replace ``XX`` by your student number.

After that, we will use the tool `bedtools`... Try:
``` 
load module mugqic/bedtools
bedtools intersect
```

Reading the tool description you should be able to intersect your peak `.bed` file with the repeatmasker `.bed` file!

## Full ChIP-seq pipeline tutorial

Here's the link to the full ChIP-seq pipeline tutorial if you're interested:

http://www.computationalgenomics.ca/tutorials/

