## Getting started

Start by loading the appropriate modules:
```
module use /cvmfs/soft.mugqic/CentOS6/modulefiles
module load mugqic/java mugqic/fastqc
```

Copy the data that you need:
```
rm -r test_data
mkdir test_data
cd test_data
cp /home/etu40/test_data/* .
```

Run the tool fastqc:
```
fastqc ENCFF361CSC_2M_ctl.fastq.gz
fastqc ENCFF837BCE_2M_ctcf.fastq.gz
```

If you do `ls` you should see the resulting .html and .zip files.

After that in a different terminal window on your local machine:
```
scp etuXX@ip31.ccs.usherbrooke.ca:/home/etuXX/test_data/*.html .
```

Replacing the 'etuXX' by your actual account number. 

After that you should be able to open the .html files locally on your browser to look at the result of fastqc.

## Alignment

We will be using the mapping tool bwa.
```
module load mugqic/bwa
```

Create shortcut for human reference sequence:
```
export hg19_ref=/cvmfs/soft.mugqic/CentOS6/genomes/species/Homo_sapiens.hg19/genome/bwa_index/Homo_sapiens.hg19.fa
```

The actual mapping command:
```
bwa aln -t 8 bwa aln -t 16 $hg19_ref ENCFF837BCE_2M_ctcf.fastq.gz > ENCFF837BCE_2M_ctcf.sai
```

Can you guess what `-t 8` means?

Next step is to create the actual alignement file:
```
bwa samse -f ENCFF837BCE_2M_ctcf.sam $hg19_ref ENCFF837BCE_2M_ctcf.sai ENCFF837BCE_2M_ctcf.fastq.gz 
```


